### DISCUSSION TITLE
New role between Admin and (collab) User

### DISCUSSION THREAD

**Discussion by npublish** [envault/envault#15] [Ideas]
**Description**
Currently there are three distinct user roles.
`Owners` can do everything on envault, `Admins` can do everthing on an App basis, while `User`s can only view a specific App they've been added to and only read the env variables in that app.

In practice, our team members who are defined as "User" and added as Collaborators to Apps need to be able to add/edit env variables as well in those Apps. That means, currently we need to add those as Admins instead.
In turn that gives them the ability to "manage" (change and even delete) Apps but curiously no ability to create new Apps.
There appears to be a huge gap between those two roles with an inconsistency about what Admins can do to "manage" Apps.
One is too restricted as a role, the other is too broad but only goes 8 yards.

If you have outsourced/external entities that need access to your .env, "Collaborator" with a readonly approach makes sense. Additions to .env must/should be communicated in this context.
For trainees or similarly untrained/new employees on the other hand it would make more sense to be able to give them roles that gives them the ability to add/edit/remove .env variables in a project they've been assigned to.
But not the ablity to create, edit or delete Apps and potentially break stuff (accidentally or on purpose).

**Describe the solution you'd like**
I propose to "fix" the Admin role and to add a new role above Collaborator/User but below Admin.
With that new role members of an App they have been assigned to are able to add/edit/remove .env variables.
But have no other permissions to manage the App itself.
While Admin users get the ability to add new Apps as well, to really manage those.

**Describe alternatives you've considered**
There are no viable alternatives as of right now.

**Comment by danharrin** [comment]
Hi! I appreciate your concerns here. Would having a more granular, customisable role system do the trick?

**Comment by npublish** [comment]
Yes, that would work as well!
